webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/components/DirectiveEditForm.scala

```diff
@@ -391,16 +391,27 @@ class DirectiveEditForm(
 
   def addFormMsg(msg: NodeSeq): Unit = formTracker.addFormError(msg)
 
-  private[this] def onFailure(): JsCmd = {
+  private[this] def onFailure(hasVariableErrors: Boolean = false): JsCmd = {
     formTracker.addFormError(error("There was a problem with your request."))
-    showErrorNotifications()
+    val cmd = if (hasVariableErrors) {
+      showVariablesErrorNotifications()
+    } else {
+      showErrorNotifications()
+    }
+    formTracker.cleanErrors
+    cmd
   }
 
   private[this] def onNothingToDo(): JsCmd = {
     formTracker.addFormError(error("There are no modifications to save."))
     showErrorNotifications()
   }
 
+  private[this] def showVariablesErrorNotifications(): JsCmd = {
+    // only replace notification container to avoid resetting the tab state
+    onFailureCallback() & Replace("notification", displayNotifications())
+  }
```

Comment:
This allows to avoid going back to the first tab when we know there are errors in the parameters tab.

If there are errors though in the first tab, they will not be reported (the parameters errors need to be fixed first)